pkg/core/pipeline/sources.go

```diff
@@ -33,7 +33,16 @@ func (p *Pipeline) RunSources() error {
 		logrus.Infof("\n%s\n", id)
 		logrus.Infof("%s\n", strings.Repeat("-", len(id)))
 
-		err = source.Run()
+		shouldRunSource := true
+		for _, parentSource := range source.Config.DependsOn {
+			if p.Sources[parentSource].Result != result.SUCCESS {
+				logrus.Warningf("Parent source[%q] did not succeed. Skipping execution of the source[%q]", parentSource, id)
+				shouldRunSource = false
```

Comment:
What do you think to break early if we identify that a "source" failed.

```suggestion
				shouldRunSource = false
				break
```

Reply:
I am wondering if we need to explicitly set the current source result to `result.Failure` if one of the dependency failed. 
Such as 
```suggestion
				shouldRunSource = false
				source.Result = result.FAILURE
```

Reply:
> I am wondering if we need to explicitly set the current source result to result.Failure if one of the dependency failed.

That feels weird since the source was not run (e.g. skipped).
Do not forget that, as for todat, the default state of a source is "skipped", which is the reason why no state change is done here (otherwise I would have set it to "skipped" instead of whatever default, because it's not run).

Reply:
> What do you think to break early if we identify that a "source" failed.

Could be discussed, but not in the scope of this PR, because it would be a breaking change in the UX behavior. Do you mind raising an issue on that topicas enhancement?